File: Behavioral/template-method/examples/name-generator/Python/NameGenerator/generators/RussianNameGenerator.py
from generators.BaseNameGenerator import BaseNameGenerator, Gender
import random


class RussianNameGenerator(BaseNameGenerator):
    __first_male_names = ["Антон", "Олег", "Роман", "Виктор"]
    __first_female_names = ["Виктория", "Светлана", "Катерина", "Ольга"]
    __surnames = ["Петров", "Семенов", "Иванов", "Борисов"]

    def generate_first_name(self, gender: Gender) -> str:
        if gender == Gender.MALE:
            return self.__first_male_names[random.randint(0, len(self.__first_male_names)-1)]
        return self.__first_female_names[random.randint(0, len(self.__first_female_names)-1)]

    def generate_surname(self, gender: Gender) -> str:
        if gender == Gender.MALE:
            return self.__surnames[random.randint(0, len(self.__surnames)-1)]
        return self.__surnames[random.randint(0, len(self.__surnames)-1)]+"а"

File: Behavioral/template-method/examples/name-generator/C#/NameGenerator/NameGenerator/Generators/RussianNameGenerator.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace NameGenerator.Generators
{
    public class RussianNameGenerator : BaseNameGenerator
    {
        private readonly string[] _firstMaleNames = new string[] { "Антон", "Олег", "Роман", "Виктор" };
        private readonly string[] _firstFemaleNames = new string[] { "Виктория", "Светлана", "Катерина", "Ольга" };
        private readonly string[] _secondNames = new string[] { "Петров", "Семенов", "Иванов", "Борисов" };

        protected override string GenerateFirstName(Gender gender)
        {
            switch (gender)
            {
                case Gender.Male:
                    return _firstMaleNames[_random.Next(0, _firstMaleNames.Length)];
                case Gender.Female:
                    return _firstFemaleNames[_random.Next(0, _firstFemaleNames.Length)];
                default:
                    throw new ArgumentOutOfRangeException();
            }

        }

        protected override string GenerateSurname(Gender gender)
        {
            switch (gender)
            {
                case Gender.Male:
                    return _secondNames[_random.Next(0, _secondNames.Length)];
                case Gender.Female:
                    return $"{_secondNames[_random.Next(0, _secondNames.Length)]}а";
                default:
                    throw new ArgumentOutOfRangeException();
            }

        }
    }
}
